### PR TITLE
CommentCompilationWarning typings

### PR DESCRIPTION
**Pull request by Legends**
**What kind of change does this PR introduce?**
CommentCompilationWarning typings

**Did you add tests for your changes?**
no

**Does this PR introduce a breaking change?**
no

**What needs to be documented once your changes are merged?**
no

### CODE REVIEW

File: lib/CommentCompilationWarning.js [hunk 1]
```diff
@@ -6,7 +6,25 @@
...
 const WebpackError = require("./WebpackError");
 
+/** @typedef {import("./Module.js")} Module */
+
+/** @typedef {Object} Position
+ *  @property {number} column
+ *  @property {number} line
+ */
+
+/** @typedef {Object} Loc
```
Comment (Legends) [inline lib/CommentCompilationWarning.js]: Should I add ` @property {number=} index` like in compareLocations.js ?

Comment (Legends) [reply lib/CommentCompilationWarning.js]: @mohsen1  The `Loc` type I have defined here is used in several locations, is there a way to not `typedef` it over and over again, but just import it from a common definition file?

Comment (sokra) [reply lib/CommentCompilationWarning.js]: You can import typedefs from other files. Maybe the best place is in ``Dependency.js`

Comment (Legends) [reply lib/CommentCompilationWarning.js]: @ooflorent Can you make a comment on sokras comment?

Comment (Legends) [reply lib/CommentCompilationWarning.js]: @mohsen1 ?

Comment (mohsen1) [reply lib/CommentCompilationWarning.js]: Importing types from  Dependency class makes sense.

Comment (Legends) [reply lib/CommentCompilationWarning.js]: Please review the changes i made, advise appreciated. I have commited this change, but I have TS errors

image

Actually, I have no clue how to do this ;-)

File: lib/Dependency.js [hunk 2]
```diff
@@ -48,4 +48,17 @@ class Dependency {
...
+ *  @property {number} column
+ *  @property {number} line
+ */
+
+/** @typedef {Object} Loc
+ *  @property {Position} start
+ *  @property {Position} end
+ */
+
+exports Position;
```
Comment (mohsen1) [inline lib/Dependency.js]: you don't need to export types

Comment (Legends) [reply lib/Dependency.js]: I'll remove the exports.. and leave everything as is.

Comment (Legends) [reply lib/Dependency.js]: @sokra done.

File: lib/CommentCompilationWarning.js [hunk 3]
```diff
@@ -6,7 +6,19 @@
 
 const WebpackError = require("./WebpackError");
 
+/** @typedef {import("./Module.js")} Module */
+
+/** @typedef {import("./Dependency.js").Position} Position */
```
Comment (mohsen1) [inline lib/CommentCompilationWarning.js]: remove unused type

Comment (Legends) [reply lib/CommentCompilationWarning.js]: done

Comment (Legends) [reply lib/CommentCompilationWarning.js]: @sokra

**Comment by sokra** [comment]
Thanks

**Pull request merged**